AmplifyPlugins/API/AWSAPICategoryPlugin/Support/Utils/GraphQLResponseDecoder+DecodeError.swift

```diff
@@ -0,0 +1,71 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Amplify
+
+extension GraphQLResponseDecoder {
+
+    static func decodeErrors(graphQLErrors: [JSONValue]) throws -> [GraphQLError] {
+        var responseErrors = [GraphQLError]()
+        for error in graphQLErrors {
+            do {
+                let responseError = try decode(graphQLErrorJSON: error)
+                responseErrors.append(responseError)
+            } catch let decodingError as DecodingError {
+                throw APIError(error: decodingError)
+            } catch {
+                throw APIError.unknown("""
+                    Unexpected failure while decoding GraphQL response containing errors:
+                    \(String(describing: graphQLErrors))
+                    """, "", error)
```

Comment:
should that second argument contain something? Does the "report to AWS" suggestion fit here?

Reply:
it looks like we already do that when exposed to the developer
```
case .unknown(_, let recoverySuggestion, _):
            return """
            \(recoverySuggestion)

            \(AmplifyErrorMessages.shouldNotHappenReportBugToAWS())
            """
```